ksql-common/src/main/java/io/confluent/ksql/function/UdfIndex.java

```diff
@@ -137,6 +138,25 @@ void addFunction(final KsqlFunction function) {
     curr.update(function, order);
   }
 
+  void addOrReplaceFunction(final KsqlFunction function) {
+    synchronized (this) {
+      if (allFunctions.containsKey(function.getArguments())) {
+        allFunctions.remove(function.getArguments());
```

Comment:
same comment as below, we need to make sure to remove the old function from the underlying `Trie`